core/src/main/java/com/netflix/iceberg/BaseTableScan.java

```diff
@@ -154,6 +155,16 @@ public InclusiveManifestEvaluator load(Integer specId) {
         }
       });
 
+  private final LoadingCache<Integer, Expression> PARTITION_EXPR_CACHE = CacheBuilder
+      .newBuilder()
```

Comment:
Sorry, this name is misleading. This is an instance field, so as soon as the scan is finished it will be garbage collected. Also, there should only ever be a few partition specs and one expression. So this cache will be a few objects at most.

Reply:
Fixed the name.